src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/GetTimelineTest.kt

```diff
@@ -0,0 +1,206 @@
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.resource.model.assertThat
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.withBody
+import java.time.LocalDate
+
+class GetTimelineTest : IntegrationTestBase() {
+
+  companion object {
+    private const val URI_TEMPLATE = "/timelines/{prisonNumber}"
+    private const val CREATE_ACTION_PLAN_URI_TEMPLATE = "/action-plans/{prisonNumber}"
+    private const val CREATE_GOAL_URI_TEMPLATE = "/action-plans/{prisonNumber}/goals"
+    private const val UPDATE_GOAL_URI_TEMPLATE = "/action-plans/{prisonNumber}/goals/{goalReference}"
+  }
+
+  @Test
+  fun `should return unauthorized given no bearer token`() {
+    webTestClient.get()
+      .uri(URI_TEMPLATE, aValidPrisonNumber())
+      .exchange()
+      .expectStatus()
+      .isUnauthorized
+  }
+
+  @Test
+  fun `should return forbidden given bearer token without required role`() {
+    // Given
+    val prisonNumber = aValidPrisonNumber()
+
+    // When
+    val response = webTestClient.get()
+      .uri(URI_TEMPLATE, prisonNumber)
+      .bearerToken(aValidTokenWithNoAuthorities(privateKey = keyPair.private))
+      .exchange()
+      .expectStatus()
+      .isForbidden
+      .returnResult(ErrorResponse::class.java)
+
+    // Then
+    val actual = response.responseBody.blockFirst()
+    assertThat(actual)
+      .hasStatus(HttpStatus.FORBIDDEN.value())
+      .hasUserMessage("Access Denied")
+      .hasDeveloperMessage("Access denied on uri=/timelines/$prisonNumber")
+  }
+
+  @Test
+  fun `should not get timeline given prisoner has no timeline`() {
+    // Given
+    val prisonNumber = aValidPrisonNumber()
+
+    // When
+    val response = webTestClient.get()
+      .uri(URI_TEMPLATE, prisonNumber)
+      .bearerToken(aValidTokenWithViewAuthority(privateKey = keyPair.private))
+      .exchange()
+      .expectStatus()
+      .isNotFound
+      .returnResult(ErrorResponse::class.java)
+
+    // Then
+    val actual = response.responseBody.blockFirst()
+    assertThat(actual)
+      .hasStatus(HttpStatus.NOT_FOUND.value())
+      .hasUserMessage("Timeline not found for prisoner [$prisonNumber]")
+  }
+
+  @Test
+  @Transactional
+  fun `should get timeline for prisoner`() {
+    // Given
+    val prisonNumber = aValidPrisonNumber()
+    val createActionPlanRequest = aValidCreateActionPlanRequest(
+      reviewDate = LocalDate.now(),
+      goals = listOf(aValidCreateGoalRequest(title = "Learn German")),
+    )
+    createActionPlan(prisonNumber, createActionPlanRequest)
+
+    // When
+    val response = webTestClient.get()
+      .uri(URI_TEMPLATE, prisonNumber)
+      .bearerToken(aValidTokenWithViewAuthority(privateKey = keyPair.private))
+      .exchange()
+      .expectStatus()
+      .isOk
+      .returnResult(TimelineResponse::class.java)
+
+    // Then
+    val actionPlan = actionPlanRepository.findByPrisonNumber(prisonNumber)
```

Comment:
Why do you retrieve the action plan from the repository here? I cant see that we use it in the assertions below?

Reply:
Sorry, just see how it's used. Ignore me 👍